src/chat.rs

```diff
@@ -2241,26 +2241,34 @@ pub async fn get_chat_msgs(
                 };
 
             Ok((
-                row.get::<_, MsgId>("id")?,
                 row.get::<_, i64>("timestamp")?,
+                row.get::<_, MsgId>("id")?,
                 !is_info_msg,
             ))
         }
     } else {
         |row: &rusqlite::Row| {
             Ok((
-                row.get::<_, MsgId>("id")?,
                 row.get::<_, i64>("timestamp")?,
+                row.get::<_, MsgId>("id")?,
                 false,
             ))
         }
     };
     let process_rows = |rows: rusqlite::MappedRows<_>| {
+        // It is faster to sort here rather than
+        // let sqlite execute an ORDER BY clause.
+        let mut sorted_rows = Vec::<(i64, MsgId, bool)>::new();
+        for row in rows {
+            let (ts, curr_id, exclude_message) = row?;
+            sorted_rows.push((ts, curr_id, exclude_message));
+        }
+        sorted_rows.sort_unstable();
```

Comment:
i did some closer look at that: it is equal than before as `sort_unstable` uses `lt` that seems to compare tuples field-by-field - is that a well-known, documented behavior?

Reply:
It's documented:
> The sequential nature of the tuple applies to its implementations of various traits. For example, in PartialOrd and Ord, the elements are compared sequentially until the first non-equal set is found.

https://doc.rust-lang.org/std/primitive.tuple.html

(Also, if this wasn't the intended way tuples are ordered, Rust would just not make tuples comparable and the code would not compile - e.g. [you can't just sort a Vec of some random structs](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cd4b43b0ebd9533c6493f14f9fb49ca8)).

Reply:
thanks for double-checking, @Hocuri!